common/scala/src/main/scala/whisk/core/entity/WhiskAction.scala

```diff
@@ -376,6 +376,24 @@ object WhiskAction extends DocumentFactory[WhiskAction] with WhiskEntityQueries[
     }
   }
 
+  override def del[Wsuper >: WhiskAction](db: ArtifactStore[Wsuper], doc: DocInfo)(
+    implicit transid: TransactionId,
+    notifier: Option[CacheChangeNotification]): Future[Boolean] = {
+    Try {
+      require(db != null, "db undefined")
+      require(doc != null, "doc undefined")
+    }.map { _ =>
+      val fa = super.del(db, doc)
+      implicit val ec = db.executionContext
+      fa.flatMap { _ =>
+        super.deleteAttachments(db, doc)
```

Comment:
as a future enhancement this could be done only when there is an attachment present and this code possibly merged into the document factory instead.

Reply:
@rabbah Any pointers on how to determine if an attachment is present ... or you meant expose a method in `DocumentFactory` like `hasAttachement` and use that and have that overridden in `WhiskAction`

Reply:
Thinking strictly about an action, today you’d know what based on the exec type and attached details. Eventually we will store all blobs as attachments so it may be inconsequential.